open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/authorize/utils/ConsentRetrievalUtil.java

```diff
@@ -256,4 +272,429 @@ public static JSONArray appendDummyAccountID() {
+            jsonElementDebtor.appendField(ConsentExtensionConstants.TITLE,
+                    ConsentExtensionConstants.DEBTOR_ACC_TITLE);
+            jsonElementDebtor.appendField(ConsentExtensionConstants.DATA_SIMPLE, debtorAccountArray);
+            consentDataJSON.add(jsonElementDebtor);
+        }
+
+        return consentDataJSON;
+
+    }
+
+
+    /**
+     * Populate VRP Details.
+     *
+     * @param controlParameters Control Parameters from the request
+     * @param consentDataJSON   Consent information object
+     */
+    private static JSONArray populateVRData(JSONObject controlParameters, JSONArray consentDataJSON) {
+
+        if (consentDataJSON == null) {
+            consentDataJSON = new JSONArray();
+        }
+
+        JSONArray paymentTypeArray = new JSONArray();
+        JSONObject jsonElementPaymentType = new JSONObject();
+
+        //Adding Payment Type
+        paymentTypeArray.add(ConsentExtensionConstants.DOMESTIC_VRP);
+        jsonElementPaymentType.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.PAYMENT_TYPE_TITLE);
+        jsonElementPaymentType.appendField(ConsentExtensionConstants.DATA_SIMPLE, paymentTypeArray);
+        consentDataJSON.add(jsonElementPaymentType);
+
+        // Constructing jsonElementValidToDataTime
+        JSONObject jsonElementValidToDateTime = new JSONObject();
+        jsonElementValidToDateTime.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.CONTROL_PARAMETER_VALID_TO_DATE_TITLE);
+        JSONArray dateControlParameterArray = new JSONArray();
+        dateControlParameterArray.add((controlParameters).
+                get(ConsentExtensionConstants.VALID_TO_DATE_TIME));
+        jsonElementValidToDateTime.appendField(ConsentExtensionConstants.DATA_SIMPLE, dateControlParameterArray);
+
+        // Constructing jsonElementValidFromDataTime
+        JSONObject jsonElementValidFromDateTime = new JSONObject();
+        jsonElementValidFromDateTime.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.CONTROL_PARAMETER_VALID_FROM_DATE_TITLE);
+        JSONArray dateTimeControlParameterArray = new JSONArray();
+        dateTimeControlParameterArray.add((controlParameters).
+                get(ConsentExtensionConstants.VALID_FROM_DATE_TIME));
+        jsonElementValidFromDateTime.appendField(ConsentExtensionConstants.DATA_SIMPLE, dateTimeControlParameterArray);
+
+        // Constructing jsonElementControlParameter
+        JSONObject jsonElementControlParameter = new JSONObject();
+        jsonElementControlParameter.appendField(ConsentExtensionConstants.TITLE,
+                ConsentExtensionConstants.CONTROL_PARAMETER_MAX_INDIVIDUAL_AMOUNT_TITLE);
+        JSONArray controlParameterArray = new JSONArray();
+
+        String formattedAmount = String.format("%s %s",
+                ((JSONObject) controlParameters.get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT))
```

Comment:
We can move `controlParameters.get(ConsentExtensionConstants.MAXIMUM_INDIVIDUAL_AMOUNT)` as common parameter and use here

Reply:
Fixed in [a6d2b0b]